pvlib/iotools/pvgis.py

```diff
@@ -383,11 +383,17 @@ def read_pvgis_hourly(filename, pvgis_format=None, map_variables=True):
                     fbuf, map_variables=map_variables)
         return pvgis_data
 
-    # raise exception if pvgis format isn't in ['csv', 'json']
-    err_msg = (
-        "pvgis format '{:s}' was unknown, must be either 'json' or 'csv'")\
-        .format(outputformat)
-    raise ValueError(err_msg)
+    elif outputformat == 'basic':
+        err_msg = "outputformat='basic' is no longer supported, please use "\
+            "outputformat='csv' instead."
+        raise ValueError(err_msg)
+
+    else:
+        # raise exception if pvgis format isn't in ['csv', 'json']
```

Comment:
move this up to the top where `outputformat` is defined to raise an exception asap.

Reply:
Moving it would duplicate code as far as I see it, so I think it's better to leave it where it is.